src/projectile.c

```diff
@@ -159,6 +159,15 @@ void projectile_set_prototype(Projectile *p, ProjPrototype *proto) {
 	p->proto = proto;
 }
 
+complex projectile_graze_size(Projectile *p) {
+	if(p->type == EnemyProj && !(p->flags & PFLAG_NOGRAZE) && p->graze_counter < 5) {
+		complex s = (p->size * 420 /* graze it */) / (2 * p->graze_counter + 1);
+		return sqrt(creal(s)) + sqrt(cimag(s)) * I;
```

Comment:
This does not keep the aspect ratio between w/h equal right? making it sqrt(w/h) in the end which will be closer to a circle.

Alternatively `return sqrt(cabs(s))* (s/cabs(s))` or shorter `return s/sqrt(cabs(s))` this won’t change the aspect ratio because it’s just dividing both by a number. but it should be checked tha